addin-proximity/app/scripts/main.js

```diff
@@ -279,13 +281,16 @@ geotab.addin.proximity = () => {
                         if (!logRecord.id) {
                             return;
                         }
-                        let dLat = toRadians(centerPoint.latitude - logRecord.latitude);
-                        let dLon = toRadians(centerPoint.longitude - logRecord.longitude);
-                        let a = Math.sin(dLat / 2.0) * Math.sin(dLat / 2.0) + Math.cos(toRadians(logRecord.latitude)) * Math.cos(toRadians(centerPoint.latitude)) * Math.sin(dLon / 2.0) * Math.sin(dLon / 2.0);
-                        let c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1 - a));
-                        logRecord.distance = 6371000.0 * c;
-                        if (logRecord.distance < maxRadius) {
-                            rtn.data.push(logRecord);
+                        // checking if the log records fall outside the selected date range
+                        if ((new Date(logRecord.dateTime) >= new Date(params.fromDate)) && (new Date(logRecord.dateTime) <= new Date(params.toDate))) {
```

Comment:
Could avoid date parsing on every itteration of loop for params.fromDate and params.toDate by doing outside the loop and reusing?

Reply:
Will fix that in the next commit.